pkg/api/validate/dynamic/dynamic.go

```diff
@@ -425,3 +452,42 @@ func uniqueSubnetSlice(slice []Subnet) []Subnet {
 	}
 	return list
 }
+
+func validateSkuLocation(ctx context.Context, location string, subscriptionID string, oc *api.OpenShiftCluster, skus []mgmtcompute.ResourceSku) error {
+	var (
+		workerProfileSku string
+		skuAvailable     bool
+	)
+
+	// Find desired worker SKU in cluster document's WorkerProfile
+	for _, workerprofile := range oc.Properties.WorkerProfiles {
+		if len(workerprofile.VMSize) != 0 {
+			workerProfileSku = string(workerprofile.VMSize)
+		} else {
+			return api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeNotFound, "", "Dynamic validation failed: Cluster WorkerProfile was not found.")
+		}
+	}
+
+	// Check to make sure desired SKU is present in cluster's install location
+	for _, sku := range skus {
+		if *sku.Name == workerProfileSku {
+			skuAvailable = true
+
+			// Fail if desired worker sku is restricted within the subscription.
+			// https://docs.microsoft.com/en-us/azure/azure-resource-manager/templates/error-sku-not-available
+			for _, restriction := range *sku.Restrictions {
+				restrictedLocations := strings.Join(*restriction.RestrictionInfo.Locations, " ")
+
+				if strings.Contains(restrictedLocations, location) && skuAvailable {
+					return api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeInvalidParameter, "", "The selected sku %v is restricted in region %v for subscription %v: restricted locations contain %v", workerProfileSku, location, subscriptionID, restrictedLocations)
+				}
+			}
+		}
+	}
```

Comment:
Thire is a bunch of checks we are doing in [here](https://github.com/Azure/ARO-RP/blob/fef8f41aa37faa8bae0e27ea896a40a549cd9d3a/pkg/env/prod.go#L278-L299). Are they not applicable here?

If they are - can we look into making something reusable?

Reply:
> I believe all we should need here is sku.Name and sku.Restrictions, since we are already getting a location-filtered SKU list.

@cadenmarchese yes, but SKU is not only about VMs. It contains SKUs for a bunch of other things (like availability sets). Are we sure that there is not, let's say availability set SKU with the same name? Probably not, but who knows.

> I was under the impression that we can't just usedv.env.VMSkus like we do here, because that pulls a list using RP subscription and we need the list pulled using the customer subscription.

Yes, we can't reuse the same data as which which we gather in [`prod.go`](https://github.com/Azure/ARO-RP/blob/fef8f41aa37faa8bae0e27ea896a40a549cd9d3a/pkg/env/prod.go#L278-L299) because it will be RP subscriptioin related, not customer subscription related. However we can reuse the code which handles/filters the data: data structure is the same, but the source of the data is different (customer sub instead of RP sub).

Reply:
@m1kola I moved the code in `NewProd` that filters the data to package `computeskus` so that it can be used in both places. Let me know if that is what you had in mind.

This change should also cover us if a SKU name is not a VM: https://github.com/Azure/ARO-RP/blob/fef8f41aa37faa8bae0e27ea896a40a549cd9d3a/pkg/env/prod.go#L284